### PR TITLE
Feat(eos_cli_config_gen): Add support additional-paths to root context of BGP using new DM

### PR DESCRIPTION
**Pull request by ccsnw**
## Change Summary

- Adding support for usage of "bgp additional-paths" in root context of BGP config
- Uses new DM as described in #3730

## Related Issue(s)

Fixes #<N/A>
Related to: #3730, #4515

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
- See updated DM in #3730 
- Adding additional-path support to a context where it isn't available yet.
- Using new DM to avoid changing that section on the future

## How to test
- Added Molecule test that can be extended to other contexts in the future

### User Checklist

- N/A

### Repository Checklist

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [X] My change requires a change to the documentation and documentation have been updated accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
converting to draft since we need to confirm some data model choices. Thanks!

**Review by ClausHolbechArista — APPROVED** [review]

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/defs_bgp_additional_paths.schema.yml [hunk 1]
```diff
@@ -17,10 +18,9 @@ $defs:
           ecmp: All paths in best path ECMP group.
           limit: Limit to n eligible paths.
           disabled: Disable sending any path.
-        valid_values: [ 'any', 'backup', 'ecmp', 'limit', 'disabled' ]
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/defs_bgp_additional_paths.schema.yml:20]: This should not be removed.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/defs_bgp_additional_paths.schema.yml [hunk 2]
```diff
@@ -17,10 +18,9 @@ $defs:
           ecmp: All paths in best path ECMP group.
           limit: Limit to n eligible paths.
           disabled: Disable sending any path.
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/defs_bgp_additional_paths.schema.yml]:
```suggestion
          disabled: Disable sending any path.
        valid_values: [ 'any', 'backup', 'ecmp', 'limit', 'disabled' ]
```

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/defs_bgp_additional_paths.schema.yml (hunk 1 above)
Comment (ccsnw) [reply python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/defs_bgp_additional_paths.schema.yml:20]: don't think I've deleted that.. else it was by accident

**Review by ClausHolbechArista — APPROVED** [review]

**Review by MaheshGSLAB — APPROVED** [review]

**Pull request merged**